parser.py

```diff
@@ -120,19 +130,43 @@ def tweet_json_to_markdown(tweet, username, archive_media_folder, output_media_f
                     else:
                         print(f'Warning: missing local file: {archive_media_path}. Using original link instead: {original_url} (expands to {original_expanded_url})')
                         markdown += f'![]({original_url})'
-        body = body.replace(original_url, markdown)
+                        html += f'<a href="{original_url}">{original_url}</a>'
+        body_markdown = body_markdown.replace(original_url, markdown)
+        body_html = body_html.replace(original_url, html)
     # make the body a quote
-    body = '> ' + '\n> '.join(body.splitlines())
+    body_markdown = '> ' + '\n> '.join(body_markdown.splitlines())
+    body_html = '<p><blockquote>' + '<br>\n'.join(body_html.splitlines()) + '</blockquote>'
     # append the original Twitter URL as a link
-    body = header + body + f'\n\n<img src="{tweet_icon_path}" width="12" /> [{timestamp_str}](https://twitter.com/{username}/status/{tweet_id_str})'
-    return timestamp, body
+    original_tweet_url = f'https://twitter.com/{username}/status/{tweet_id_str}'
+    body_markdown = header_markdown + body_markdown + f'\n\n<img src="{tweet_icon_path}" width="12" /> [{timestamp_str}]({original_tweet_url})'
+    body_html = header_html + body_html + f'<a href="{original_tweet_url}"><img src="{tweet_icon_path}" width="12" />&nbsp;{timestamp_str}</a></p>'
+    return timestamp, body_markdown, body_html
 
 def main():
 
     input_folder = '.'
     output_media_folder_name = 'media/'
     tweet_icon_path = f'{output_media_folder_name}tweet.ico'
     media_sources_filename = 'media_sources.txt'
+    output_html_filename = 'TweetArchive.html'
+
+    HTML = """\
+<!doctype html>
+<html lang="en">
+<head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <link rel="stylesheet"
+          href="https://unpkg.com/@picocss/pico@latest/css/pico.min.css">
```

Comment:
Can we make a simple CSS to avoid requiring an online one?

Reply:
The fallback when offline (tested in flight mode) is fine, being just plain HTML.